src/server.rs

```diff
@@ -109,27 +109,37 @@ macro_rules! messages {
             $($method_name$(($method_arg))*,)*
         }
         fn parse_message(input: &str) -> Result<ServerMessage, ParseError>  {
-            let ls_command: serde_json::Value = serde_json::from_str(input).unwrap();
+            let ls_command: serde_json::Value = match serde_json::from_str(input) {
+                Ok(inp) => inp,
+                Err(_) => return Err(ParseError::new(ErrorKind::InvalidData, "unable to parse json.", None)) 
+            };
```

Comment:
okay, I'll take a look. I still find the macro syntax hard to read, but maybe this is a good excuse to dig in a little bit. 👍